torch/csrc/cuda/python_nccl.cpp

```diff
@@ -0,0 +1,305 @@
+#include "nccl.h"
+#include "torch/csrc/THP.h"
+#include "torch/csrc/Types.h"
+#include "torch/csrc/DynamicTypes.h"
+#include "torch/csrc/cuda/THCP.h"
+
+#include "torch/csrc/cuda/nccl.h"
+
+#include <nccl.h>
+#include <sstream>
+#include <unordered_map>
+
+using namespace at;
+using namespace torch::cuda::nccl::detail;
+
+static inline void CHECK(ncclResult_t status) {
+  if (status != ncclSuccess) {
+    std::stringstream err;
+    err << "NCCL Error " << status << ": " << ncclGetErrorString(status);
+    throw std::runtime_error(err.str());
+  }
+}
+
+static const char* COMM_CAPSULE_NAME = "torch.cuda.nccl.Communicator";
+
+PyObject * THCPModule_nccl_version(PyObject *self, PyObject *args) {
+#if defined(NCCL_MAJOR)
```

Comment:
Should this call `version()` from `torch/csrc/cuda/nccl.cpp`? Looks like the logic is duplicated.